### PR TITLE
Pull request for Capstone React-Redux Project

### PR DESCRIPTION
**Pull request by chuaindia**
This is a Capstone project where different crypto currencies are fetched from API.

The project contains two pages:

**-Home page :**

1. When the page loads, the web app shows the list of categories that could be filtered out by some parameter, for example by the category name.
2. Along with the category name, it will display some numeric values per category (The coins)that come from the API.
3. When the user clicks (or taps) on a category item, the application navigates to the details page.

**-Details page**

1. In the details page, the web app retrieves data from the API to show detailed data in the category.
2. When the user clicks on the "Back" button (<), the user navigates to the home page.

The demonstration of the project is given in the attached video:  
https://www.loom.com/share/e00e6d11de7c470583a7c56fc11a4fc4

The test page is also attached here:

https://user-images.githubusercontent.com/106825373/204974914-3e0fe17c-b2cf-46fd-a3a4-208d47863b68.png

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,127 @@
...
+## Overview
 
-If you aren't satisfied with the build tool and configuration choices, you can `eject` at any time. This command will remove the single build dependency from your project.
+The Loom video Link : https://www.loom.com/share/e4b38fb8217245fb94b363cb6f302a44 
 
-Instead, it will copy all the configuration files and the transitive dependencies (webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except `eject` will still work, but they will point to the copied scripts so you can tweak them. At this point you're on your own.
+<!-- LIVE DEMO -->
 
-You don't have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn't feel obligated to use this feature. However we understand that this tool wouldn't be useful if you couldn't customize it when you are ready for it.
+## 🚀 Live Demo <a name="live-demo"></a>
```
Comment (Davidosky007) [inline README.md:45]: * Kindly provide a working link to your deployed project as it's a [ requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=Deploy%20the%20project%20and%20test%20for%20final%20details.)

File: README.md [hunk 2]
```diff
@@ -1,70 +1,127 @@
...
-### `npm run eject`
+- **Fetch Cryptocurrencies[Render Detail Pages Dynamically]**
+- **Fetch all data corresponding to a cryptocurrency from corresponding API**
+- **Filtering by different coins**
 
-**Note: this is a one-way operation. Once you `eject`, you can't go back!**
+## Overview
 
-If you aren't satisfied with the build tool and configuration choices, you can `eject` at any time. This command will remove the single build dependency from your project.
+The Loom video Link : https://www.loom.com/share/e4b38fb8217245fb94b363cb6f302a44
```
Comment (Davidosky007) [inline README.md]: * Kindly fulfill these requirements below 👇 as your video recording doesn't cover all [these requirements ](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=Once%20you%20have,very%20proud%20of.)

> Once you have finished the development of the project, you should record a video presenting the features of the project you built. It is a video with a maximum length of 5 minutes. The content of the video should include:

a description of the project.
a demo of the project features.
you should also highlight some interesting piece of code or something you built that you are very proud of.

File: src/components/Detail.css [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  display: flex;
+  justify-content: center;
+  border: none;
+  padding-top: 2rem;
+}
+
+.Row td {
+  padding: 15px;
+  font-size: 20px;
+  font-family: monospace, sans-serif;
```
Comment (Davidosky007) [inline src/components/Detail.css]: * Kindly follow the design guidelines which include font families( Lato, Gill sans) as its [required ](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=You%20should%20follow,space%20between%20elements.)

File: src/components/Home.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import Coinsdisplay from './coins';
+import { displayCoin } from '../redux/home/home';
+import './Home.css';
+
+const Coin = () => {
+  const allcoin = useSelector((state) => state.covid);
```
Comment (Davidosky007) [inline src/components/Home.js:9]: * Your project lacks the filter using given parameters functionality, for example 

> When the page loads, the web app shows the list of categories that could be filtered out by some parameter, let's say by the category name.
* Kindly fix this 🙏

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @chuaindia 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Good looking readme file 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Detail.css (hunk 3 above)
Comment (chuaindia) [reply src/components/Detail.css]: Included font-family as required.

File: README.md (hunk 2 above)
Comment (chuaindia) [reply README.md]: I am extremely sorry. I put a previous link. I am attaching the loom video corresponding to this project.

https://www.loom.com/share/e4b38fb8217245fb94b363cb6f302a44

File: README.md (hunk 1 above)
Comment (chuaindia) [reply README.md:45]: I have made the proper deployment with Netlify. The link is attached here for your reference:

https://capstone-react-redux-chuaindia.netlify.app/

File: src/components/Home.js (hunk 4 above)
Comment (chuaindia) [reply src/components/Home.js:9]: I made the filter on different coins.

Comment (chuaindia) [reply src/components/Home.js:9]: It is based on the key imported as v4uuid from uuid.
So I did not make any other filter.

File: src/components/Home.css [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  background-color: #d45586;
+}
+
+table {
+  padding: 0;
+  border: 1px solid #989898;
+}
+
+.Header {
+  font-family: monospace, sans-serif;
```
Comment (brytebee) [inline src/components/Home.css]: - Kindly update this to the [design guideline](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)) font.

Your Design | Template
:---: | :---:
image | image

File: src/components/Home.css [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  border: 2px black;
+}
+
+.CoinContainer > tr {
+  flex: 50%;
+}
+
+td {
+  color: white;
+  font-family: monospace, sans-serif;
```
Comment (brytebee) [inline src/components/Home.css]: Same as above.

File: src/components/Detail.css (hunk 3 above)
Comment (brytebee) [reply src/components/Detail.css]: Not fixed.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @chuaindia👋, 

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️
- Good video presentation with enthusiasm and what you're proud of.✔️
- A working link is attached as suggested by the previous reviewer.✔️

##

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/components/Home.css (hunk 5 above)
Comment (chuaindia) [reply src/components/Home.css]: Done. It will be Lato, Gill Sans.

File: src/components/Home.css (hunk 6 above)
Comment (chuaindia) [reply src/components/Home.css]: Done. It will be Lato, Gill Sans.

File: src/components/Detail.css (hunk 3 above)
Comment (chuaindia) [reply src/components/Detail.css]: I did it.

**Comment by chuaindia** [comment]
I changed all the font-families. Please review it now. I did not disable it. I provided it according to my VS code editor.

**Comment by chuaindia** [comment]
I fixed all the font-family as required by the Reviewer.

**Review by brytebee — APPROVED** [review]
Hello, @chuaindia👋 ,

------

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

------

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by chuaindia** [comment]
I am really very very much thankful to you.

**Pull request merged**